### ISSUE TITLE
Data Flow Diagram for OpenEMR

### ISSUE THREAD

**Issue by Lord-Tiger** [Lord-Tiger/CYBR8420_Fall24#15]

**Comment by asongne** [comment]
![image](https://github.com/user-attachments/assets/e490d5fa-1657-4a3d-99a1-349522769b03)

Do you think we can integrate this to your diagram?

**Comment by Lord-Tiger** [comment]
Nice! Do we want to introduce a boundary somewhere here?

**Comment by asongne** [comment]
Yes i think we need to. the only place i see fit is adding the authentication piece

**Comment by asongne** [comment]
For this phase of our project, we focused on building a Data Flow Diagram (DFD) for our use case. We held a meeting with our professor to seek guidance, and it was decided that we would concentrate on the authentication component, as it encompasses many of the system's processes. Following the meeting, we developed the initial DFD and reconvened on Sunday to further refine and expand it.

The primary takeaway from this activity has been recognizing the value of threat modeling software. It highlighted potential security issues we had not previously considered, prompting us to conduct a deeper analysis to categorize and determine the relevance of various threats. Additionally, we needed to research historical and current issues or gaps within the software to understand how similar vulnerabilities have been addressed in the past.

In summary, this activity has encouraged us to proactively consider security flaws during the design process and leverage the threat modeling tool to uncover additional vulnerabilities, ultimately strengthening the system’s overall security
